TESTS/network/interface/networkinterface_status.cpp

```diff
@@ -36,6 +36,10 @@ nsapi_connection_status_t statuses[status_buffer_size];
 
 void status_cb(nsapi_event_t event, intptr_t value)
 {
+    if ((event >= NSAPI_EVENT_CELLULAR_STATUS_BASE) && (event <= NSAPI_EVENT_CELLULAR_STATUS_END)) {
```

Comment:
If we assume that this is status change, we should compare that value is exactly `NSAPI_EVENT_CONNECTION_STATUS_CHANGE`

So change the clause to `if (event != NSAPI_EVENT_CONNECTION_STATUS_CHANGE)`

Reply:
this way: `if (event != NSAPI_EVENT_CONNECTION_STATUS_CHANGE)`
test will not assert for values outside the valid range for nsapi_event values. Is that preferable?

Reply:
Hmm... at least it would not break if we extend the `nsapi_event_t` enum in the future.